server.go

```diff
@@ -363,6 +435,8 @@ func (s *Server) ListenAndServe(ctx context.Context, hostport string) error {
 	}
 	s.log.Print("Listening on " + hostport)
 
+	go s.monitorHandler()
```

Comment:
nit: Just to be on the safe side, you could use the syntax used below and have the goroutine use a wait group.
```go
wg.Add(1)
go func() {
    defer wg.Done()
    s.monitorHandler()
}()
```